game/src/engine/AI/AIService.h

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <GLFW/glfw3.h>
+
+#include <glm/glm.hpp>
+#include <map>
+#include <stack>
+#include <vector>
+
+#include "../service/Service.h"
+
+//----------------
+// Node
+//----------------
+
+class Node
+{
+  public:
+    Node(unsigned int id, glm::vec3 v0, glm::vec3 v1, glm::vec3 v2);
+    ~Node();
+
+    glm::vec3 Get_Centroid(Node* node);
+
+    unsigned int id_;                    // used to define every triangle
+    glm::vec3 v0_, v1_, v2_, centroid_;  // to define every point we need for
+                                         // our navmesh algorithm to work
+
+    std::vector<std::pair<float, Node*>>* connections_;
+
+  private:
+};
```

Comment:
Can we make this a nested public class of Navmesh? Probably don't want something generic like `Node` in the global namespace